examples/mongo_odm/demo.cpp

```diff
@@ -0,0 +1,221 @@
+    r2.cuisine = "Middle Eastern";
+    r2.grades.push_back({sysclock_t::now(), "B", 11});
+    r2.grades.push_back({sysclock_t::now(), "C", 17});
+    r2.name = "Maoz";
+    r2.restaurant_id = "41704621";
+
+    // // insert restaurants into collection
+    auto res = restaurants.insert_one(r1);
+    res = restaurants.insert_one(r2);
+
+    // get all restaurants in collection
+    {
+        std::cout << "Finding all restaurants in the collection..." << std::endl;
+        mongo_odm::deserializing_cursor<Restaurant> cur = restaurants.find({});
+        for (Restaurant r : cur) {
+            std::cout << "Restaurant: " << r.name << std::endl;
+        }
+        std::cout << std::endl;
+    }
+
+    // Query by top-level field
+    {
+        std::cout << "Querying by cuisine (top-level field):" << std::endl;
+        bsoncxx::builder::stream::document filter;
+        filter << "cuisine"
+               << "Italian";
+        mongocxx::stdx::optional<Restaurant> r_italian = restaurants.find_one(filter.view());
+        if (r_italian) {
+            std::cout << "Restaurant: " << r_italian->name << " (" << r_italian->cuisine << ")"
+                      << std::endl;
+        } else {
+            std::cout << "No matching restaurants were found." << std::endl;
+        }
+        std::cout << std::endl;
+    }
+
+    // Query by embedded field
+    {
+        std::cout << "Querying by street (embedded field):" << std::endl;
+        bsoncxx::builder::stream::document filter;
+        filter << "address.street"
+               << "7 Avenue";
+        auto cur = restaurants.find(filter.view());
+        for (Restaurant r : cur) {
+            std::cout << "Restaurant: " << r.name << " " << r.address << std::endl;
+        }
+        std::cout << std::endl;
+    }
+
+    // Pass options to find()
+    {
+        std::cout << "Pass options to find(): " << std::endl;
+        mongocxx::options::find opts;
+        bsoncxx::builder::stream::document order_builder;
+        order_builder << "borough" << -1 << "address.zipcode" << 1;
+        opts.sort(order_builder.view());
+
+        auto cur = restaurants.find({}, opts);
+        for (Restaurant r : cur) {
+            std::cout << "Restaurant: " << r.name << ", " << r.borough << " " << r.address.zipcode
+                      << std::endl;
+        }
+    }
+
+    // Insert several objects at once
+
+    Restaurant r3;
+    r3.name = "Morris Park Bake Shop";
+    r3.cuisine = "Bakery";
+    r3.borough = "Bronx";
+    r3.address = {"1007", "Morris Park Ave", "10462", -73.856077, 40.848447};
+    r3.grades.push_back({make_time_point(1393804800000), "A", 2});
+    r3.grades.push_back({make_time_point(1378857600000), "A", 6});
+    r3.grades.push_back({make_time_point(1358985600000), "A", 10});
+    r3.grades.push_back({make_time_point(1322006400000), "A", 9});
+    r3.grades.push_back({make_time_point(1299715200000), "B", 14});
+    r3.restaurant_id = "30075445";
+
+    //     Restaurant r4;
```

Comment:
The demo's not finished yet, I was working on it while the CMake / cereal issues came up.